cmd/taskmasterd/yaml.go

```diff
@@ -280,6 +288,12 @@ func (program *ProgramYaml) Validate(args ProgramYamlValidateArgs) (ProgramConfi
 	}
 
 	if program.Workingdir != nil {
+		if !isASCII(*program.Workingdir) {
```

Comment:
One thing to note if we accept only ASCII paths is that any accented character will be refused.
Apparently the solely character that is not permitted in a UNIX path is the `nul` character: https://stackoverflow.com/a/537833/11780994.
As being permissive in this area of the application does not lead to crashes or bugs, we should probably just create a function `isPathValid` that checks if the path contains a `nul` character.

To me we could use the same approach for `Name` field (that we must validate because if `Stdout` or `Stderr` properties are not filled, they will be computed using `Name` property), `Cmd`, `Stdout` and `Stderr`.